packages/eas-cli/src/credentials/ios/actions/RemovePushKey.ts

```diff
@@ -0,0 +1,52 @@
+import { ApplePushKeyFragment } from '../../../graphql/generated';
+import Log from '../../../log';
+import { confirmAsync } from '../../../prompts';
+import { Account } from '../../../user/Account';
+import { Context } from '../../context';
+import { selectPushKeyAsync } from './PushKeyUtils';
+
+export class SelectAndRemovePushKey {
+  constructor(private account: Account) {}
+
+  async runAsync(ctx: Context): Promise<void> {
+    const selected = await selectPushKeyAsync(ctx, this.account);
+    if (selected) {
+      await new RemovePushKey(this.account, selected).runAsync(ctx);
+      Log.succeed('Removed push key');
+      Log.newLine();
+    }
+  }
+}
+
+export class RemovePushKey {
+  constructor(private account: Account, private pushKey: ApplePushKeyFragment) {}
+
+  public async runAsync(ctx: Context): Promise<void> {
+    if (ctx.nonInteractive) {
+      throw new Error(`Cannot remove push keys in non-interactive mode`);
+    }
+
+    const confirm = await confirmAsync({
+      message: `Removing this push key will disable push notifications for projects that rely on it. Do you want to continue?`,
```

Comment:
shouldn't we show what project are using that specific push key?

Reply:
Yup, i'll need to expose this in our graphql api in www first -- i've added a todo and a linear task for myself